### PR TITLE
PHPUnit Cleanup - Fixes #2053

### LINKED ISSUE #2053: PHPUnit Doesn't Clear Caches
Before submitting your issue, please make sure that you've checked the checkboxes below.

- [x] I am running the [latest release](https://github.com/CachetHQ/Cachet/releases/latest) version of Cachet.
- [x] I am running at least PHP 5.5.9. *You can check this by running `php -v`.*
- [x] I have ran `rm -rf bootstrap/cache/*`.

### Expected behaviour

After running `phpunit`, I expect my environment to not be mutated.

### Actual behaviour

PHPUnit bootstraps it's environment and never clears it's cache once completed (application stays in testing).

### Steps to reproduce

1. Have a default installation (pre-setup and seeded)
2. Run `vendor/bin/phpunit`
3. Go to your site, notice you're now on `/setup`
4. Run `artisan env` and notice you're now in `testing`

### PR DESCRIPTION
**Pull request by ConnorVG**
See #2053

### CODE REVIEW

File: tests/TestListener.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * This is the test listener class.
+ *
+ * @author Connor S. Parks <connor@connorvg.tv>
+ */
+class TestListener extends PHPUnit_Framework_BaseTestListener
+{
+    /**
+     * A test suite ended.
+     *
+     * @since Method available since Release 2.2.0
```
Comment (GrahamCampbell) [inline tests/TestListener.php]: Please remove this.

File: tests/TestListener.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     *
+     * @return void
+     */
+    public function endTestSuite(PHPUnit_Framework_TestSuite $suite)
+    {
+        if ($suite->getName() !== 'Cachet Test Suite') {
+            return;
+        }
+
+        foreach (glob(__DIR__.'/../bootstrap/cache/*.php') as $file) {
```
Comment (GrahamCampbell) [inline tests/TestListener.php]: We should clear the cachet cache folder too.

**Comment by GrahamCampbell** [comment]
Please send to the 2.4 branch.

**Comment by ConnorVG** [comment]
10/10

**Pull request closed without merging**